blog/content/post/2022-03-15-fobnail_3rd_phase.md

```diff
@@ -0,0 +1,202 @@
+blob, which enables us to modify it when such a need arises.
+
+Reference Integrity Manifest (RIM) is supposed to be created, signed and
+provided by OEM, OS and/or software vendors, depending on which PCRs are touched
+by those entities. Their format is described in multiple specifications. Some of
+them are contradictory, others demand things that are physically impossible
+(measurements must include all implemented banks for at least 8 PCRs, but have
+to be returned in TPML_DIGEST which can hold max 8 digests). We also decided to
+skip event log for the time being for the sake of easier testing in CI and
+because of buggy firmware on host running attester...
+
+For easier parsing on Fobnail side we put PCR values in CBOR format. We also
+skipped metadata from RIM since we already sent it separately. By doing these
+changes we basically created another standard:
+
+[![Standards](https://imgs.xkcd.com/comics/standards.png)](https://xkcd.com/927/)
+
+# Building
+
+Build system was updated for easier and faster development. It now includes
+packages required to build TPM simulator. This opens up a possibility of
+CI-testing full solution, instead of just build tests for each component
+separately.
+
+### Cloning
+
+```bash
+$ git clone https://github.com/fobnail/fobnail-sdk.git
+$ git clone https://github.com/fobnail/fobnail-attester.git --recurse-submodules
+$ git clone https://github.com/fobnail/fobnail.git --recurse-submodules
+```
+
+Components should always work with each other when all repositories are cloned
+at the same time, but for extra safety, these are commits used at the time of
+writing this blog post:
+
+* SDK: `53f19086c993 2022-03-08|Fix build problems on nRF target`
+* Attester: `85e8ab442e9f 2022-03-17|docker.sh: review fixes;
+  docker/entrypoint.sh: fix permissions issue`
+* Fobnail: `9a8a404f9e5f 2022-03-16|tools/lfs/src/main.rs: add option to format
+  flash before doing command`
+
+### Building and installing SDK
+
+Fobnail SDK is used to build code for Fobnail token, i.e. Verifier part of
+attestation process.
+
+```bash
+$ cd fobnail-sdk
+# This takes some time, but it is one-time operation until SDK is updated
+$ ./build.sh
+# Feel free to use different directory or name, as long as it is in $PATH
+$ ln -s $(readlink -f ./run-fobnail-sdk.sh) ~/bin/run-fobnail-sdk.sh
+```
+
+### Building Attester's Docker image
+
+Attester requires access to SMBIOS tables to read metadata (platform serial
+number, manufacturer etc.) so as of now it has to be started with different set
+of permissions. For this reason, another Docker container is used for building
+and running Attester. This container also manages building every component
+through one script to make it as easy to use as possible. You just have to point
+it to Verifier's code and everything else is done by that script.
+
+```bash
+$ cd fobnail-attester
+$ docker build -t fobnail/fobnail-attester .
+```
```

Comment:
Please move this code into section below. It is a bit out of context here.

Reply:
This is part of "Building Attester's Docker image", it would be out of context in next section. I will try to merge these into one.

Reply:
@krystian-hebel ok, missed that, I thought it was in Building and installing SDK